tools/gmon/gmon/clients/utils.py

```diff
@@ -0,0 +1,47 @@
+"""
+`utils.py`
+Util functions.
+"""
+
+from google.protobuf.json_format import MessageToDict
+
+
+def decorate(decorator, methods=[]):
+    def inner(cls):
+        for attr in cls.__dict__:  # there's propably a better way to do this
```

Comment:
if it's empty, wouldn't we just not decorate the class instead ?

Reply:
I think I'm being a bit nit picky for the purpose of this tool, but in general we should be careful decorating "all" methods of a class

IMO yes you should specify the methods to be decorated for two reasons:
1) 
> Explicit better than implicit
[PEP20 "Zen of Python"](https://www.python.org/dev/peps/pep-0020/#:~:text=Explicit%20is%20better%20than%20implicit,Flat%20is%20better%20than%20nested.)

2) you may accidentally end up decorating properties (which look like methods) or some methods you don't typically think about to e.g. `__init__` `__eq__`. 

Typically monkey patching is "at your own risk" but It will be very confusing if someone monkey patches a method into your class why their method is behaving differently (e.g. because your decorator will decorate their method).

I think the general principal is decorators should be applied pretty explicitly.